src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -239,6 +241,10 @@ public Void call() throws Exception {
                         return null;
                     }
 
+                    for (FlowExecutionListener listener : ExtensionList.lookup(FlowExecutionListener.class)) {
+                        listener.onRunning(execution);
```

Comment:
This is going to be called many times during a build: every time we reënter the CPS VM thread. That does not match the expectations given by its Javadoc, which is one call at build start time and one call per running build per Jenkins restart.